lib/routemap.h

```diff
@@ -123,6 +123,9 @@ struct route_map_rule_cmd {
 
 	/* Free allocated value by func_compile (). */
 	void (*func_free)(void *);
+
+	/** To get the rule key after Compilation **/
+	const char* (*func_get_rmap_rule_key)(void *val);
```

Comment:
I'm curious what if the value isn't char? I'm trying to find other cases except for this one because it's more generic "hook" than just for this case :)